### ISSUE TITLE
php 8.2: dynamic properties deprecated, multiple warnings

### ISSUE THREAD

**Issue by torvista** [zencart/zencart#5103]
[Deprecate dynamic properties](https://wiki.php.net/rfc/deprecate_dynamic_properties)

Thought I'd try the php 8.2beta: 94 warnings on entering the admin, and all for the same error but different files.
It would be good if someone could confirm if anything other than just declaring these variables is necessary....

This is useful, there seem to be only a couple more things to review:
https://stitcher.io/blog/new-in-php-82

**Comment by lat9** [comment]
A copy (or a section) of the logged warnings would help.

**Comment by torvista** [comment]
Well, here's a section of the 1200+ lines.
The format is as generated by my modified error_log: https://github.com/zencart/zencart/issues/5102

> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 ADMIN/includes/classes/AdminRequestSanitizer.php(79): zen_debug_error_handler()
> #1 ADMIN/includes/classes/AdminRequestSanitizer.php(62): AdminRequestSanitizer->__construct()
> #2 ADMIN/includes/extra_datafiles/advshipper_sanitization.php(4): AdminRequestSanitizer::getInstance()
> #3 ADMIN/includes/init_includes/init_file_db_names.php(45): require('D://OneDrive//Bus...')
> #4 ADMIN/includes/application_bootstrap.php(156): require('D://OneDrive//Bus...')
> #5 ADMIN/index.php(9): require_once('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property AdminRequestSanitizer::$arrayName is deprecated in ADMIN/includes/classes/AdminRequestSanitizer.php on line 79.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/db/mysql/query_factory.php(61): zen_debug_error_handler()
> #1 ADMIN/includes/init_includes/init_database.php(18): queryFactory->connect()
> #2 ADMIN/includes/application_bootstrap.php(157): require('D://OneDrive//Bus...')
> #3 ADMIN/index.php(9): require_once('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property queryFactory::$pConnect is deprecated in includes/classes/db/mysql/query_factory.php on line 61.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/PluginManager.php(16): zen_debug_error_handler()
> #1 ADMIN/includes/application_bootstrap.php(160): Zencart/PluginManager/PluginManager->__construct()
> #2 ADMIN/index.php(9): require_once('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property Zencart\PluginManager\PluginManager::$pluginControl is deprecated in includes/classes/PluginManager.php on line 16.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/PluginManager.php(17): zen_debug_error_handler()
> #1 ADMIN/includes/application_bootstrap.php(160): Zencart/PluginManager/PluginManager->__construct()
> #2 ADMIN/index.php(9): require_once('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property Zencart\PluginManager\PluginManager::$pluginControlVersion is deprecated in includes/classes/PluginManager.php on line 17.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/PluginManager.php(16): zen_debug_error_handler()
> #1 ADMIN/includes/application_bootstrap.php(162): Zencart/PluginManager/PluginManager->__construct()
> #2 ADMIN/index.php(9): require_once('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property Zencart\PluginManager\PluginManager::$pluginControl is deprecated in includes/classes/PluginManager.php on line 16.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/PluginManager.php(17): zen_debug_error_handler()
> #1 ADMIN/includes/application_bootstrap.php(162): Zencart/PluginManager/PluginManager->__construct()
> #2 ADMIN/index.php(9): require_once('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property Zencart\PluginManager\PluginManager::$pluginControlVersion is deprecated in includes/classes/PluginManager.php on line 17.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/ResourceLoaders/PageLoader.php(19): zen_debug_error_handler()
> #1 ADMIN/includes/application_bootstrap.php(164): Zencart/PageLoader/PageLoader->init()
> #2 ADMIN/index.php(9): require_once('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property Zencart\PageLoader\PageLoader::$installedPlugins is deprecated in includes/classes/ResourceLoaders/PageLoader.php on line 19.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/ResourceLoaders/PageLoader.php(20): zen_debug_error_handler()
> #1 ADMIN/includes/application_bootstrap.php(164): Zencart/PageLoader/PageLoader->init()
> #2 ADMIN/index.php(9): require_once('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property Zencart\PageLoader\PageLoader::$mainPage is deprecated in includes/classes/ResourceLoaders/PageLoader.php on line 20.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/ResourceLoaders/PageLoader.php(21): zen_debug_error_handler()
> #1 ADMIN/includes/application_bootstrap.php(164): Zencart/PageLoader/PageLoader->init()
> #2 ADMIN/index.php(9): require_once('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property Zencart\PageLoader\PageLoader::$fileSystem is deprecated in includes/classes/ResourceLoaders/PageLoader.php on line 21.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/InitSystem.php(25): zen_debug_error_handler()
> #1 ADMIN/includes/application_top.php(35): Zencart/InitSystem/InitSystem->__construct()
> #2 ADMIN/home.php(9): require('D://OneDrive//Bus...')
> #3 ADMIN/index.php(21): require('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property Zencart\InitSystem\InitSystem::$installedPlugins is deprecated in includes/classes/InitSystem.php on line 25.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/InitSystem.php(26): zen_debug_error_handler()
> #1 ADMIN/includes/application_top.php(35): Zencart/InitSystem/InitSystem->__construct()
> #2 ADMIN/home.php(9): require('D://OneDrive//Bus...')
> #3 ADMIN/index.php(21): require('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property Zencart\InitSystem\InitSystem::$debug is deprecated in includes/classes/InitSystem.php on line 26.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/InitSystem.php(27): zen_debug_error_handler()
> #1 ADMIN/includes/application_top.php(35): Zencart/InitSystem/InitSystem->__construct()
> #2 ADMIN/home.php(9): require('D://OneDrive//Bus...')
> #3 ADMIN/index.php(21): require('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property Zencart\InitSystem\InitSystem::$debugList is deprecated in includes/classes/InitSystem.php on line 27.
> 
> [10-Aug-2022 18:31:43 Europe/Madrid]
> ADMIN
> Request URI: /tienda/ADMIN/
> IP address: 127.0.0.1
> #0 includes/classes/InitSystem.php(28): zen_debug_error_handler()
> #1 ADMIN/includes/application_top.php(35): Zencart/InitSystem/InitSystem->__construct()
> #2 ADMIN/home.php(9): require('D://OneDrive//Bus...')
> #3 ADMIN/index.php(21): require('D://OneDrive//Bus...')
> --> PHP Deprecated: Creation of dynamic property Zencart\InitSystem\InitSystem::$actionList is deprecated in includes/classes/InitSystem.php on line 28.

**Comment by lat9** [comment]
Yikes, this is going to be a major PITA.  See this (https://php.watch/versions/8.2/dynamic-properties-deprecated) link for additional details.

**Comment by torvista** [comment]
I don't know about other IDE's but phpStorm has been pointing out these things for a while.

**Comment by scottcwilson** [comment]
Thanks for reporting (sob).
I will create a new tag for tracking PHP 8.2 issues.

**Comment by brittainmark** [comment]
I tested a change to server info for HTML5 complience in php 8.2 all i did to get it to work was declare the variable as Public in the classes and all was well.

**Comment by torvista** [comment]
Did you have strict reporting enabled? I always have that on for development.

**Comment by lat9** [comment]
> I tested a change to server info for HTML5 complience in php 8.2 all i did to get it to work was declare the variable as Public in the classes and all was well.

It's not the complexity of the change, it's its broad reach (i.e. the number of modules that need to be &mdash; at a minimum &mdash; reviewed) that makes this a PITA.

**Comment by torvista** [comment]
One PR per file?

**Comment by zcwilt** [comment]
So currently our testsuite doesn't test against PHP8.2
We could run tests, but not make them obligatory
given 8.2 is in beta

Why at this point do we need to look to do php8.2 compatibility when there is so much more to do.

**Comment by lat9** [comment]
> One PR per file?

That would be good.

> So currently our testsuite doesn't test against PHP8.2 We could run tests, but not make them obligatory given 8.2 is in beta
> 
> Why at this point do we need to look to do php8.2 compatibility when there is so much more to do.

It's relatively low-hanging fruit and needs to be done eventually.

**Comment by brittainmark** [comment]
> 

Sorry I did not. Will set it and see if it makes a difference.

**Comment by brittainmark** [comment]
Strict reporting made no difference. Setting to public removed the depreciated.

**Comment by zcwilt** [comment]
https://github.com/zencart/zencart/pull/5143

Have added a 8.2 test but set it not to fail MR's 

It's passing at the moment, but guess we aren't hitting any endpoints where it would fail.

**Comment by brittainmark** [comment]
Is this a mindless task. Is it sufficient to add var property for the list of errors. 
for example, object_info.php add
```
var $configuration_id,
        $configuration_title,
        $configuration_value,
        $configuration_key,
        $use_function,
        $configuration_description,
        $date_added,
        $last_modified,
        $set_function;
```
If this is sufficient, then I would be happy to go through all the classes and create pr's for them. 
If it needs more thinking about, then probably not a job for me.
Finally, If you are happy for me to do it, how many do you want in a day?

**Comment by torvista** [comment]
> Have added a 8.2 test but set it not to fail MR's

What are MRs?

**Comment by scottcwilson** [comment]
Merge Requests.

**Comment by lat9** [comment]
> Is this a mindless task. Is it sufficient to add var property for the list of errors. for example, object_info.php add
> 
> ```
> var $configuration_id,
>         $configuration_title,
>         $configuration_value,
>         $configuration_key,
>         $use_function,
>         $configuration_description,
>         $date_added,
>         $last_modified,
>         $set_function;
> ```
> 
> If this is sufficient, then I would be happy to go through all the classes and create pr's for them. If it needs more thinking about, then probably not a job for me. Finally, If you are happy for me to do it, how many do you want in a day?

In my mind &mdash; but not necessarily everyone's &mdash; the opportunity should also be taken to change those `var` statements to a more current `public`, `protected` or `private` depending on the variable.  For those in the example posted, it's sufficient to change `var` to `public` and add all the missing ones.

**Comment by brittainmark** [comment]
> In my mind — but not necessarily everyone's — the opportunity should also be taken to change those var statements to a more current public, protected or private depending on the variable. For those in the example posted, it's sufficient to change var to public and add all the missing ones.

What determines if it should be protected of private?

**Comment by brittainmark** [comment]
Have started going through object_info.php and think I have found an issue.

If someone (me) writes a plugin that has fields that needs to be displayed/edited on the product screen. Then the field name needs to be added to the declarations in object_info.php this would be true for any database changes or anywhere that object_info is used to set up the class with the appropriate data.

For example, I want to add products_location to the product screen. admin/includes/classes/product/collect_info.php has notifiers that allow extra fields to be added  $zco_notifier->notify('NOTIFY_ADMIN_PRODUCT_COLLECT_INFO_EXTRA_INPUTS', $pInfo, $extra_product_inputs) the issue is how do you let the objectInfo class know what the variable is called to allow it to be declared.

Does this mean that there will need to be a complete rethink on how plugins work for php8.2.

**Comment by brittainmark** [comment]
Just so we know how potentially big a task this is I have included a task list of all the classes and their locations.
I have not included vendor classes.
- [ ] **AdminArraysLanguageLoader** ->includes/classes/ResourceLoaders/AdminArraysLanguageLoader.php:
- [ ] **AdminCatalogUrlGenerationTest** ->not_for_release/testFramework/unittests/testsHtmlOutput/AdminCatalogUrlGenerationTest.php:
- [ ] **AdminDuskTestCase** ->not_for_release/testFramework/Browser/AdminDuskTestCase.php:
- [ ] **AdminFilesLanguageLoader** ->includes/classes/ResourceLoaders/AdminFilesLanguageLoader.php:
- [ ] **AdminLoggingTest** ->not_for_release/testFramework/unittests/testsNotifiers/AdminLoggingTest.php:
- [ ] **AdminLoginPageTest** ->not_for_release/testFramework/Browser/AdminLoginPageTest.php:
- [ ] **AdminNotifications** ->admin/includes/classes/AdminNotifications.php:
- [ ] **AdminNotificationsTest** ->not_for_release/testFramework/unittests/testsAdminNotifications/AdminNotificationsTest.php:
- [ ] **AdminRequestSanitizer** ->admin/includes/classes/AdminRequestSanitizer.php:
- [ ] **AdminSanitizationTest** ->not_for_release/testFramework/unittests/testsAdminSanitization/AdminSanitizationTest.php:
- [ ] **AdminUrlGenerationTest** ->not_for_release/testFramework/unittests/testsHtmlOutput/AdminUrlGenerationTest.php:
- [ ] **ArraysLanguageLoader** ->includes/classes/ResourceLoaders/ArraysLanguageLoader.php:
- [ ] **AttributeLookupsTest** ->not_for_release/testFramework/unittests/testsSundry/AttributeLookupsTest.php:
- [ ] **authorizenet** ->includes/modules/payment/authorizenet.php:
- [ ] **authorizenet_aim** ->includes/modules/payment/authorizenet_aim.php:
- [ ] **base** ->includes/classes/class.base.php:
- [ ] **BaseController** ->includes/classes/ViewBuilders/BaseController.php:
- [ ] **baseFilter** ->includes/classes/Filters/baseFilter.php:
- [ ] **BaseLanguageLoader** ->includes/classes/ResourceLoaders/BaseLanguageLoader.php:
- [ ] **BasePluginInstaller** ->includes/classes/PluginSupport/BasePluginInstaller.php:
- [ ] **box** ->admin/includes/classes/box.php:
- [ ] **boxTableBlock** ->admin/includes/classes/table_block.php:
- [ ] **breadcrumb** ->includes/classes/breadcrumb.php:
- [ ] **BreadcrumbTest** ->not_for_release/testFramework/unittests/testsSundry/BreadcrumbTest.php:
- [ ] **cache** ->includes/classes/cache.php:
- [ ] **CardinalXMLParser** ->includes/modules/payment/paypaldp.php:
- [ ] **CatalogArraysLanguageLoader** ->includes/classes/ResourceLoaders/CatalogArraysLanguageLoader.php:
- [ ] **CatalogFilesLanguageLoader** ->includes/classes/ResourceLoaders/CatalogFilesLanguageLoader.php:
- [ ] **CatalogUrlGenerationTest** ->not_for_release/testFramework/unittests/testsHtmlOutput/CatalogUrlGenerationTest.php:
- [ ] **CategoriesTreeTest** ->not_for_release/testFramework/unittests/testsCategories/CategoriesTreeTest.php:
- [ ] **category_tree** ->includes/classes/category_tree.php:
- [ ] **categoryPulldown** ->includes/classes/categoryPulldown.php:
- [ ] **cc_validation** ->includes/classes/cc_validation.php:
- [ ] **cod** ->includes/modules/payment/cod.php:
- [ ] **configurationValidation** ->admin/includes/classes/configurationValidation.php:
- [ ] **currencies** ->includes/classes/currencies.php:
- [ ] **Customer** ->includes/classes/Customer.php:
- [ ] **DataTableDataSource** ->includes/classes/ViewBuilders/DataTableDataSource.php:
- [ ] **DerivedItemsManager** ->includes/classes/ViewBuilders/DerivedItemsManager.php:
- [ ] **DiscountCouponsTest** ->not_for_release/testFramework/unittests/testsDiscountCoupon/DiscountCouponsTest.php:
- [ ] **DuskTestCase** ->not_for_release/testFramework/Browser/DuskTestCase.php:
- [ ] **EmailValidationTest** ->not_for_release/testFramework/unittests/testsSundry/EmailValidationTest.php:
- [ ] **EventDto** ->includes/classes/EventDto.php:
- [ ] **FilesLanguageLoader** ->includes/classes/ResourceLoaders/FilesLanguageLoader.php:
- [ ] **FileSystem** ->includes/classes/FileSystem.php:
- [ ] **FilterFactory** ->includes/classes/Filters/FilterFactory.php:
- [ ] **FilterManager** ->includes/classes/Filters/FilterManager.php:
- [ ] **firstdata_hco** ->includes/modules/payment/firstdata_hco.php:
- [ ] **flat** ->includes/modules/shipping/flat.php:
- [ ] **FmodRoundTest** ->not_for_release/testFramework/unittests/testsSundry/FmodRoundTest.php:
- [ ] **freecharger** ->includes/modules/payment/freecharger.php:
- [ ] **freeoptions** ->includes/modules/shipping/freeoptions.php:
- [ ] **freeshipper** ->includes/modules/shipping/freeshipper.php:
- [ ] **HtmlEntityRecurseTest** ->not_for_release/testFramework/unittests/testsSundry/HtmlEntityRecurseTest.php:
- [ ] **httpClient** ->includes/classes/http_client.php:
- [ ] **InitSystem** ->includes/classes/InitSystem.php:
- [ ] **InstallDuskTestCase** ->not_for_release/testFramework/Browser/InstallDuskTestCase.php:
- [ ] **Installer** ->includes/classes/PluginSupport/Installer.php:
- [ ] **InstallerFactory** ->includes/classes/PluginSupport/InstallerFactory.php:
- [ ] **IssetorArrayTest** ->not_for_release/testFramework/unittests/testsSundry/IssetorArrayTest.php:
- [ ] **item** ->includes/modules/shipping/item.php:
- [ ] **language** ->includes/classes/language.php:
- [ ] **LanguageLoader** ->includes/classes/ResourceLoaders/LanguageLoader.php:
- [ ] **LanguageLoaderFactory** ->includes/classes/ResourceLoaders/LanguageLoaderFactory.php:
- [ ] **LanguageManager** ->zc_install_junk/includes/classes/LanguageManager.php:
- [ ] **LaravelPaginator** ->admin/includes/classes/LaravelPaginator.php:
- [ ] **messageStack** ->admin/includes/classes/message_stack.php:
- [ ] **messageStack** ->includes/classes/message_stack.php:
- [ ] **Mobile_Detect** ->includes/classes/Mobile_Detect.php:
- [ ] **moneyorder** ->includes/modules/payment/moneyorder.php:
- [ ] **MultipleAdminPagesTest** ->not_for_release/testFramework/Browser/MultipleAdminPagesTest.php:
- [ ] **navigationHistory** ->includes/classes/navigation_history.php:
- [ ] **newsletter** ->admin/includes/modules/newsletters/newsletter.php:
- [ ] **notifier** ->includes/classes/class.notifier.php:
- [ ] **NotifierUpdateHandlersTest** ->not_for_release/testFramework/unittests/testsNotifiers/NotifierUpdateHandlersTest.php:
- [x] **objectInfo** ->admin/includes/classes/object_info.php:
- [ ] **ObserverAliasingTest** ->not_for_release/testFramework/unittests/testsNotifiers/ObserverAliasingTest.php:
- [ ] **order** ->includes/classes/order.php:
- [ ] **order_total** ->includes/classes/order_total.php:
- [ ] **ot_cod_fee** ->includes/modules/order_total/ot_cod_fee.php:
- [ ] **ot_coupon** ->includes/modules/order_total/ot_coupon.php:
- [ ] **ot_group_pricing** ->includes/modules/order_total/ot_group_pricing.php:
- [ ] **ot_gv** ->includes/modules/order_total/ot_gv.php:
- [ ] **ot_loworderfee** ->includes/modules/order_total/ot_loworderfee.php:
- [ ] **ot_shipping** ->includes/modules/order_total/ot_shipping.php:
- [ ] **ot_subtotal** ->includes/modules/order_total/ot_subtotal.php:
- [ ] **ot_tax** ->includes/modules/order_total/ot_tax.php:
- [ ] **ot_total** ->includes/modules/order_total/ot_total.php:
- [ ] **PageLoader** ->includes/classes/ResourceLoaders/PageLoader.php:
- [ ] **Paginator** ->admin/includes/classes/Paginator.php:
- [ ] **PasswordHashTest** ->not_for_release/testFramework/unittests/testsPasswordHash/PasswordHashTest.php:
- [ ] **payment** ->includes/classes/payment.php:
- [ ] **paypal** ->includes/modules/payment/paypal.php:
- [ ] **paypal_curl** ->includes/modules/payment/paypal/paypal_curl.php:
- [ ] **paypaldp** ->includes/modules/payment/paypaldp.php:
- [ ] **paypalwpp** ->includes/modules/payment/paypalwpp.php:
- [ ] **perweightunit** ->includes/modules/shipping/perweightunit.php:
- [ ] **PluginErrorContainer** ->includes/classes/PluginSupport/PluginErrorContainer.php:
- [ ] **PluginInstaller** ->not_for_release/zc_plugins/ModList/1.4.0/Installer/PluginInstaller.php:
- [ ] **PluginInstaller** ->not_for_release/zc_plugins/ModList/1.4.1/Installer/PluginInstaller.php:
- [ ] **PluginInstallerException** ->includes/classes/Exceptions/PluginInstallerException.php:
- [ ] **PluginManager** ->includes/classes/PluginManager.php:
- [ ] **PluginManagerController** ->includes/classes/ViewBuilders/PluginManagerController.php:
- [ ] **PluginManagerDataSource** ->includes/classes/ViewBuilders/PluginManagerDataSource.php:
- [ ] **product_notification** ->admin/includes/modules/newsletters/product_notification.php:
- [ ] **ProductConfigurationSwitch** ->includes/classes/ProductConfigurationSwitch.php:
- [ ] **productOptionsPulldown** ->includes/classes/productOptionsPulldown.php:
- [ ] **productPulldown** ->includes/classes/productPulldown.php:
- [ ] **products** ->includes/classes/products.php:
- [ ] **products_viewed_counter** ->includes/classes/observers/class.products_viewed_counter.php:
- [ ] **pulldown** ->includes/classes/pulldown.php:
- [ ] **QueryBuilder** ->includes/classes/QueryBuilder.php:
- [ ] **QueryCache** ->includes/classes/query_cache.php:
- [ ] **queryFactory** ->includes/classes/db/mysql/query_factory.php:
- [ ] **queryFactoryMeta** ->includes/classes/db/mysql/query_factory.php:
- [ ] **queryFactoryResult** ->includes/classes/db/mysql/query_factory.php:
- [ ] **Request** ->includes/classes/Request.php:
- [ ] **ScriptedInstaller** ->includes/classes/PluginSupport/ScriptedInstaller.php:
- [ ] **ScriptedInstaller** ->zc_plugins/DisplayLogs/v3.0.0/Installer/ScriptedInstaller.php:
- [ ] **ScriptedInstallerFactory** ->includes/classes/PluginSupport/ScriptedInstallerFactory.php:
- [ ] **SelectWhereFilter** ->includes/classes/Filters/SelectWhereFilter.php:
- [ ] **shipping** ->includes/classes/shipping.php:
- [ ] **shoppingCart** ->includes/classes/shopping_cart.php:
- [ ] **SideboxFinder** ->includes/classes/ResourceLoaders/SideboxFinder.php:
- [ ] **SimpleDataFormatter** ->includes/classes/ViewBuilders/SimpleDataFormatter.php:
- [ ] **sniffer** ->includes/classes/sniffer.php:
- [ ] **splitPageResults** ->admin/includes/classes/split_page_results.php:
- [ ] **splitPageResults** ->includes/classes/split_page_results.php:
- [ ] **SqlPatchInstaller** ->includes/classes/PluginSupport/SqlPatchInstaller.php:
- [ ] **square** ->includes/modules/payment/square.php:
- [ ] **statsSalesReportGraph** ->admin/includes/classes/stats_sales_report_graph.php:
- [ ] **storepickup** ->includes/modules/shipping/storepickup.php:
- [ ] **systemChecker** ->zc_install_junk/includes/classes/class.systemChecker.php:
- [ ] **table** ->includes/modules/shipping/table.php:
- [ ] **tableBox** ->includes/classes/boxes.php:
- [ ] **TableViewDefinition** ->includes/classes/ViewBuilders/TableViewDefinition.php:
- [ ] **template_func** ->includes/classes/template_func.php:
- [ ] **upload** ->includes/classes/upload.php:
- [ ] **VersionServer** ->admin/includes/classes/VersionServer.php:
- [ ] **WhosOnline** ->admin/includes/classes/WhosOnline.php:
- [ ] **zcAjaxAdminNotifications** ->includes/classes/ajax/zcAjaxAdminNotifications.php:
- [ ] **zcAjaxPayment** ->includes/classes/ajax/zcAjaxPayment.php:
- [ ] **zcConfigureFileReader** ->zc_install_junk/includes/classes/class.zcConfigureFileReader.php:
- [ ] **zcConfigureFileWriter** ->zc_install_junk/includes/classes/class.zcConfigureFileWriter.php:
- [ ] **zcDatabaseInstaller** ->zc_install_junk/includes/classes/class.zcDatabaseInstaller.php:
- [ ] **zcDate** ->includes/classes/zcDate.php:
- [ ] **zcDiscountCouponTest** ->not_for_release/testFramework/unittests/support/zcDiscountCouponTest.php:
- [ ] **ZcInstallTest** ->not_for_release/testFramework/Browser/ZcInstallTest.php:
- [ ] **zcNotifierBaseAliasTestObject** ->not_for_release/testFramework/unittests/support/zcNotifierBaseAliasTestObject.php:
- [ ] **zcNotifierTestObserver** ->not_for_release/testFramework/unittests/support/zcNotifierTestObserver.php:
- [ ] **zcNotifierTraitAliasTestObject** ->not_for_release/testFramework/unittests/support/zcNotifierTraitAliasTestObject.php:
- [ ] **zcObserverAliasTestObject** ->not_for_release/testFramework/unittests/support/zcObserverAliasTestObject.php:
- [ ] **zcObserverDownloadsViaAws** ->includes/classes/observers/auto.downloads_via_aws.php:
- [ ] **zcObserverDownloadsViaRedirect** ->includes/classes/observers/auto.downloads_via_redirect.php:
- [ ] **zcObserverDownloadsViaStreaming** ->includes/classes/observers/auto.downloads_via_streaming.php:
- [ ] **zcObserverDownloadsViaUrl** ->includes/classes/observers/auto.downloads_via_url.php:
- [ ] **zcObserverLogEventListener** ->admin/includes/classes/class.admin.zcObserverLogEventListener.php:
- [ ] **zcObserverLogWriterDatabase** ->admin/includes/classes/class.admin.zcObserverLogWriterDatabase.php:
- [ ] **zcObserverLogWriterTextfile** ->admin/includes/classes/class.admin.zcObserverLogWriterTextfile.php:
- [ ] **zcObserverNonCaptchaObserver** ->includes/classes/observers/auto.non_captcha_observer.php:
- [ ] **zcObserverProductMusicObserver** ->admin/includes/classes/observers/auto.ProductMusicObserver.php:
- [ ] **zcObserverTableControllerTest** ->admin/includes/classes/observers/auto.TableControllerTest.php:
- [ ] **zcPassword** ->includes/classes/class.zcPassword.php:
- [ ] **zcRegistry** ->zc_install_junk/includes/classes/class.zcRegistry.php:
- [ ] **zcTestCase** ->not_for_release/testFramework/unittests/support/zcTestCase.php:
- [ ] **zcURLTestObserver** ->not_for_release/testFramework/unittests/support/zcURLTestObserver.php:
- [ ] **zen_categories_ul_generator** ->includes/classes/categories_ul_generator.php:
- [ ] **zen_SiteMapTree** ->includes/classes/site_map.php:
- [ ] **ZenCartConnectCreatePaymentRequest** ->includes/modules/payment/square_support/ZenCartConnectCreatePaymentRequest.php:
- [ ] **zones** ->includes/modules/shipping/zones.php:

**Comment by lat9** [comment]
@brittainmark, thanks for doing this.  It'll be a super-big help to understand when we're finished!

**Comment by torvista** [comment]
How is this being done?
Just working back from the errors?
Examining each file / using an IDE to inspect them?

While at first glance fixing php8.2-related things seemed to be jumping the gun, it is exuding real bugs and I think demonstrating exactly why these changes get introduced in php, to improve the code.

@brittainmark, thanks for running with this. Your current input puts the rest of us to shame.

**Comment by brittainmark** [comment]
It is a mixture of looking for errors/warnings and understanding what is going on in the code. I had to do paypal from first principles as I could not get it to install on my 8,2 server.

The process I followed was.
1. Choose a class.
2. Check for internal parameters being used by the class, if any, see if they appear to be the correct type. 
3. Find all php files using that class.
4. Run each occurrence.
   1. Check reported warnings.
   6.  Add public for all items referenced in the class by the outside module (not a class)
I think that was pretty much it.

With objectInfo it was relatively straight forward as every thing had to be public as it was needed in the calling routines.

It has shown up the big issue of customization that I do not know how to get around. I should probably raise another issue for this to be discussed separately.

I wanted to get one done so people could comment on it to see if the style was OK.

The process has thrown up a number of issues as I test as I go and so have found a number of faults.  It has also allowed me to gain a better understanding of Zen cart, although some of it is still not known to me. I suspect the number of faults I find will slow down as I have been through most of admin now.

**Comment by brittainmark** [comment]
It took about 3-4 days to do objectInfo
It's taken about 1/2 day to do table_block see #5157.  Please comment on it as I need to make sure my OO understanding is OK.

**Comment by lat9** [comment]
Be very careful when a class (e.g. order.php) uses notifications.  For those classes, I'll suggest that all variables be defined as `public`; otherwise, existing observers that might be interrogating/updating those variables will fail due to an access error.

**Comment by torvista** [comment]
I'll note php8.10 is now complaining about template_func, query_factory and zcDate, so this work is proving to be necessary now.

**Comment by brittainmark** [comment]
I am working my way through looking at the warnings generated in 8.2. This way, I prioritise the ones that have warnings.  Hopefully, I picked up those (8.1) warnings in the changes #5188, #5187 and #5186. I'll keep plugging away.
I think storefront is nearly there. The ones I see most there are messageStack, category_tree and splitPageResults. I'll do those next, then keep working through admin. I will probably not do the not_for_release ones as I do not know how to run the test framework.

**Comment by torvista** [comment]
Doh, my bad, using 8.2rc1 by  mistake.

**Comment by brittainmark** [comment]
No problem. I'll keep plugging away. Just takes the pressure off.

**Comment by scottcwilson** [comment]
Fantastic progress.  I was worried we'd barely be ready for just PHP 8.0 a few months back, now we're much further ahead.

**Comment by scottcwilson** [comment]
This is coming along nicely.  Placing an order can now be done under PHP8.2/ZC158 with no logs except for issue #5223.

Plugins of course will be another matter. :(

**Comment by brittainmark** [comment]
Working my way through plugins at the moment. It's like wading thought treacle. Need a break every so often on something else to let my brain relax again.

Still need to get paypal and authorize done for complete set. Waiting on #5060 there is still an issue with tax when you have not yet got the customer address

As for adding in the #[AllowDynamicProperties] i suggest that we leave that until everything processed then add to classes we believe require it.

**Comment by scottcwilson** [comment]
As soon as Cindy gives the signal I will get #5060 merged.

**Comment by scottcwilson** [comment]
No we're really doing fantastic, thank you so much for your help. 

My focus now will be writing notes for plugin authors about what THEY need to do to get on the PHP 8.2 train. :)

**Comment by brittainmark** [comment]
Apart from PayPal and Authorize i think that the classes causing errors done.
I will look at Authorize next and see if I can get a test site set up.

If you find any I have missed, feel free to post issues, and I'll look at them.

**Comment by scottcwilson** [comment]
If you can get them to just not error when the page Admin > Modules > Payment is brought up, that's half way there.  We might need to wait until stores go live to get the rest of the issues fixed, because as you know you have to hit each of the actual code paths.

**Comment by brittainmark** [comment]
All Admin classes have now been looked at.

**Comment by scottcwilson** [comment]
Fantastic. You have really helped us move Zen Cart forward - thank you.

**Comment by brittainmark** [comment]
Class tablebox has been depreciated, so has not been changed.

**Comment by brittainmark** [comment]
Cannot find any use of class httpClient so i have not changed it

**Comment by brittainmark** [comment]
includes/classes/QueryBuilder.php: there are lots of notifiers, but noting is passed except the notifier name and all parameters are protected. Have not changed.

**Comment by brittainmark** [comment]
I think that is everything except paypalwpp

**Comment by lat9** [comment]
paypalwpp update just posted (#5300).

**Comment by scottcwilson** [comment]
Sit tight for now so we can get 1.5.8 out.  Thanks so much for everything you have done!

**Comment by scottcwilson** [comment]
I think this ticket can be closed.

**Comment by brittainmark** [comment]
Do you want a new issue for the test framework? I suggest leave it until 1.5.8 released, then raise a new issue if needed.

**Comment by scottcwilson** [comment]
I'm not so worried about issues for code that's not released.
